core/src/main/scala/org/http4s/internal/parsing/Rfc8941.scala

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.internal.parsing
+
+import cats.parse.Parser
+import cats.parse.Parser0
+import cats.parse.Rfc5234
+
+/** Parsers for the rules defined in RFC8941
+  *
+  * @see [[https://datatracker.ietf.org/doc/html/rfc8941]]
+  */
+private[http4s] object Rfc8941 {
+
+  /* sf-integer = ["-"] 1*15DIGIT
+   */
+  val sfInteger: Parser[Long] = {
+    import Parser.char, Rfc5234.digit
+    val pos = digit.rep(1, 15).string
+    val neg = (char('-') *> pos).map(s => s"-$s")
+    (pos | neg).map(_.toLong)
+  }
+
+  /* sf-decimal = ["-"] 1*12DIGIT "." 1*3DIGIT
+   */
+  val sfDecimal: Parser[BigDecimal] = {
+    import Parser.char, Rfc5234.digit
+    val num = digit.rep(1, 12).string
+    val dec = digit.rep(1, 3).string
+    val pos = (num ~ (char('.') *> dec)).map(t => s"${t._1}.${t._2}")
+    val neg = (char('-') *> pos).map(s => s"-$s")
+    (pos | neg).map(BigDecimal(_))
+  }
+
+  /* sf-string = DQUOTE *chr DQUOTE
+   * chr       = unescaped / escaped
+   * unescaped = %x20-21 / %x23-5B / %x5D-7E
+   * escaped   = "\" ( DQUOTE / "\" )
+   */
+  val sfString: Parser[String] = {
+    import Parser.{charIn, string}, Rfc5234.dquote
+    val escaped = string("\\\\") | string("\\\"")
```

Comment:
Shouldn't we unescape these when parsing and re-escape them when rendering?

Reply:
Done.